### ISSUE TITLE
Task name should be optional

### ISSUE THREAD

**Issue by jacobg** [aertje/cloud-tasks-emulator#45]
Per Google docs, the task name is optional:
https://cloud.google.com/tasks/docs/reference/rest/v2/projects.locations.queues.tasks#Task.FIELDS.name

However, this emulator fails if the task name is not specified:
https://github.com/aertje/cloud-tasks-emulator/blob/314e4e35863ff82162027e9df07ee2e32382b34b/task.go#L31-L39

```
task emulator output: panic: runtime error: index out of range [1] with length 0
task emulator output: 
task emulator output: goroutine 1060 [running]:
task emulator output: main.parseTaskName(0xc0007ce080, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
task emulator output: /app/task.go:34 +0x16e
task emulator output: main.dispatch(0xc0007ce001, 0xc0007ce080, 0x0)
task emulator output: /app/task.go:300 +0x137
task emulator output: main.(*Task).doDispatch(0xc000186380, 0xc0007ce201)
task emulator output: /app/task.go:364 +0x38
task emulator output: main.(*Task).Attempt(0xc000186380)
task emulator output: /app/task.go:374 +0x3e
task emulator output: main.(*Queue).runWorker(0xc0004b2770)
task emulator output: /app/queue.go:127 +0x3f
task emulator output: created by main.(*Queue).runWorkers
task emulator output: /app/queue.go:119 +0x48
task emulator output: 

```

**Comment by aertje** [comment]
Hi @jacobg, thanks for reporting; As per the sample in the README, the following all works for me:

`client.create_task(task={'http_request': {'http_method': 'GET', 'url': 'https://www.google.com'}}, parent=queue_name)`
`client.create_task(task={'name': None, 'http_request': {'http_method': 'GET', 'url': 'https://www.google.com'}}, parent=queue_name)`
`client.create_task(task={'name': "", 'http_request': {'http_method': 'GET', 'url': 'https://www.google.com'}}, parent=queue_name)`

And always assigns a unique name automatically (e.g. `"/projects/my-sandbox/locations/us-central1/queues/test/tasks/15352856648520921629"`).

Can you please provide me the command you used in your client to create the task in your client that leads to this error, as I'm not able to reproduce.

Thanks, Aert.

**Comment by jacobg** [comment]
Hi @aertje 

Thanks for the reply.

I'd had worked around it by setting a task name in my code that creates the task, but now when I reverted the change to try to repro the original error, it's not occurring anymore. I don't know what changed. One theory is that when I started using this emulator I was also in the middle of updating python `google-cloud-tasks` from version 1.x to the latest 2.6, and perhaps the error occurred on version 1.x client. Not sure. In any event, I'll close this issue for now, and if I stumble upon the error again, I can reopen it.

Take care.
